aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -2243,6 +2243,74 @@ Tensor nonzero_cpu(const Tensor& self) {
   return result;
 }
 
+Tensor& nonzero_static_out_cpu(
+    const Tensor& self,
+    int64_t size,
+    int64_t fill_value,
+    Tensor& result) {
+  // Sanity check on input
+  TORCH_CHECK(size >= 0, "'size' must be an non-negative integer");
+  int64_t ndim = self.dim();
+  TORCH_CHECK(ndim > 0, "Rank must be positive");
+  TORCH_CHECK(
+      result.scalar_type() == kLong,
+      "nonzero_static: Expected out tensor to have scalar type Long "
+      "but got scalar type",
+      result.scalar_type());
+  TORCH_CHECK(
+      result.dim() == 2,
+      "nonzero_static: Expected out tensor to be a 2D tensor but got a ",
+      result.dim(),
+      "D tensor");
+  TORCH_CHECK(
+      result.size(0) == size && result.size(1) == ndim,
+      "nonzero_static: Expected out tensor to have Size([",
+      size,
+      ", ",
+      ndim,
+      "]) but got Size([",
+      result.size(0),
+      ", ",
+      result.size(1),
+      "]) ");
+  at::assert_no_internal_overlap(result);
+  at::assert_no_overlap(result, self);
+
+  // Delegate call to regular nonzero to get a data-dependent output
+  auto dyn_result = nonzero_cpu(self);
+  int64_t num_nonzeros = dyn_result.size(0);
+  int64_t copy_len = std::min(size, num_nonzeros);
+  // Copy the dynamic result to the fixed-size tensor
+  result.narrow(0, 0, copy_len).copy_(dyn_result.narrow(0, 0, copy_len));
+  if (size > copy_len) {
+    // Pad result with `fill_value`
+    result.narrow(0, copy_len, size - copy_len).fill_(fill_value);
+  }
+  return result;
+}
+
+Tensor nonzero_static_cpu(
+    const Tensor& self,
+    int64_t size,
+    int64_t fill_value) {
+  int64_t ndim = self.dim();
+  // Check if self is not 0-d tensor
+  TORCH_CHECK(ndim > 0, "Rank must be positive");
+  // Check if self is not 0-d tensor
+  TORCH_CHECK(size >= 0, "'size' must be an non-negative integer");
+  if (size == 0) {
+    // Return an empty tensor with size=(0, 1)
+    return at::empty({0, ndim}, ScalarType::Long);
+  }
+  // Allocate fixed-size out tensor
+  auto result = at::full(
+      {size, self.dim()},
+      fill_value,
```

Comment:
You explicitly pad inside the out function, so it's not necessary to fill here. Use at::empty instead.